crates/oxc_linter/src/rules/react/iframe_missing_sandbox.rs

```diff
@@ -0,0 +1,227 @@
+use oxc_ast::ast::{
+    Argument, Expression, JSXAttributeItem, JSXAttributeValue, JSXElementName, ObjectProperty,
+    ObjectPropertyKind, StringLiteral,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::{phf_set, Set};
+
+use crate::utils::{get_prop_value, has_jsx_prop_ignore_case, is_create_element_call};
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn missing_sandbox_prop(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("An iframe element is missing a sandbox attribute")
+        .with_help("Add a `sandbox` attribute to the `iframe` element.")
+        .with_label(span)
+}
+
+fn invalid_sandbox_prop(span: Span, value: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("An iframe element defines a sandbox attribute with invalid value: {value}"))
+        .with_help("Change the `sandbox` attribute to one of the allowed values: ``, `allow-downloads-without-user-activation`, `allow-downloads`, `allow-forms`, `allow-modals`, `allow-orientation-lock`, `allow-pointer-lock`, `allow-popups`, `allow-popups-to-escape-sandbox`, `allow-presentation`, `allow-same-origin`, `allow-scripts`, `allow-storage-access-by-user-activation`, `allow-top-navigation`, `allow-top-navigation-by-user-activation`.")
+        .with_label(span)
+}
+
+fn invalid_sandbox_combination_prop(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("An `iframe` element defines a sandbox attribute with both allow-scripts and allow-same-origin which is invalid")
+        .with_label(span)
```

Comment:
[non-blocking comment]: we could add some help text here like: ``Remove `allow-scripts` or `allow-same-origin`.``